pkg/controller/healthcheck/add.go

```diff
@@ -58,6 +68,16 @@ func RegisterHealthChecks(mgr manager.Manager, opts healthcheck.DefaultAddArgs)
 				ConditionType: string(gardencorev1beta1.ShootControlPlaneHealthy),
 				HealthCheck:   general.NewSeedDeploymentHealthChecker(azure.CloudControllerManagerName),
 			},
+			{
+				ConditionType: string(gardencorev1beta1.ShootControlPlaneHealthy),
+				HealthCheck:   general.NewSeedDeploymentHealthChecker(azure.CSIControllerName),
```

Comment:
In the azure case the csi driver deploy names are `csi-driver-controller-disk` and `csi-driver-controller-file`.

```suggestion
				HealthCheck:   general.NewSeedDeploymentHealthChecker(azure.CSIControllerDiskName),

```

Reply:
nice catch, thanks. fixed.